src/docker/executor.cpp

```diff
@@ -318,7 +329,26 @@ class DockerExecutorProcess : public ProtobufProcess<DockerExecutorProcess>
               }
             };
 
-            if (container.ipAddress.isSome()) {
+            bool useCNI = false;
+#ifdef __linux__  
+            if (network_cni_config_dir.size() > 0 && network_cni_plugins_dir.size() > 0) {
```

Comment:
use `!x.empty()`? Or it seems like these should be `Option<string>` instead to explicitly type in the optionality and avoid empty string == unset

Also, does it make sense for only one to be set? If not, should there be some validation somewhere for this?